src/Microsoft.AspNetCore.Blazor/Rendering/Renderer.cs

```diff
@@ -217,13 +221,31 @@ private void RenderInExistingBatch(RenderQueueEntry renderQueueEntry)
             }
         }
 
-        private void RemoveEventHandlerIds(ArrayRange<int> eventHandlerIds)
+        private void RemoveEventHandlerIds(ArrayRange<int> eventHandlerIds, Task afterTask)
         {
-            var array = eventHandlerIds.Array;
-            var count = eventHandlerIds.Count;
-            for (var i = 0; i < count; i++)
+            if (eventHandlerIds.Count == 0)
             {
-                _eventBindings.Remove(array[i]);
+                return;
+            }
+
+            if (afterTask.IsCompleted)
```

Comment:
For WebAssembly execution (where the UI updates are synchronous), this condition is always true, so the changes in this PR have no effect on either behavior or perf.